File: 백준/Silver/5800. 성적 통계/성적 통계.c
#include <stdio.h>
#include <stdlib.h>
int compare(const void *pa, const void *pb) {
    const int a = *(int *)pa;
    const int b = *(int *)pb;
    return b - a;
}
int main(void) {
    int k, n, maxdiff, maxn, minn, *arr;
    scanf("%d", &k);
    for (int i = 0; i < k; i++) {
        scanf("%d", &n);
        maxdiff = 0, maxn = 0, minn = 100;
        arr = (int *)malloc(sizeof(int) * n);
        for (int j = 0; j < n; j++) {
            scanf("%d", &arr[j]);
            if (maxn < arr[j])
                maxn = arr[j];
            if (minn > arr[j])
                minn = arr[j];
        }
        qsort(arr, n, sizeof(int), compare);
        for (int j = 0; j < n - 1; j++)
            if (maxdiff < arr[j] - arr[j + 1])
                maxdiff = arr[j] - arr[j + 1];
        printf("Class %d\n", i + 1);
        printf("Max %d, Min %d, Largest gap %d\n", maxn, minn, maxdiff);
    }
    return 0;
}

File: 백준/Silver/5800. 성적 통계/성적 통계.py
import sys
input=lambda:sys.stdin.readline().rstrip()
for i in range(int(input())):
    n,*arr=map(int,input().split())
    arr.sort(reverse=True)
    maxn=0
    for j in range(n-1):
        if maxn<arr[j]-arr[j+1]:
            maxn=arr[j]-arr[j+1]
    print(f"Class {i+1}")
    print(f"Max {max(arr)}, Min {min(arr)}, Largest gap {maxn}")